activesupport/test/current_attributes_test.rb

```diff
@@ -0,0 +1,80 @@
+require "abstract_unit"
+require "active_support/current_attributes"
+
+Person = Struct.new(:name, :time_zone)
+
+class Current < ActiveSupport::CurrentAttributes
+  attribute :world, :account, :person
+
+  resets { Time.zone = "UTC" }
+
+  def account=(account)
+    attributes[:account] = account
+    self.person = "#{account}'s person"
+  end
+
+  def person=(person)
+    attributes[:person] = person
```

Comment:
We make it work in AR/AMo, where they similarly belong to the current class.

It just means that instead of defining the built-in methods in the class directly, we put them into an anonymous module included for that purpose.

It's cleverer than we *need* to be, but I think it's simple enough that it's worth the gain in "do what I mean" expressiveness. It'll help people who do expect it to work, and I can't think of any way it'd manage to unpleasantly surprise someone who doesn't.